watson/cli.py

```diff
@@ -265,14 +285,36 @@ def status(watson):
     ))
 
 
+_SHORTCUT_OPTIONS = ['year', 'month', 'week', 'day']
+
+
 @cli.command()
-@click.option('-f', '--from', 'from_', type=Date,
+@click.option('-f', '--from', 'from_', cls=MutuallyExclusiveOption, type=Date,
```

Comment:
I did – it was a first version of my code. But when I use `Feature Switches`, it isn't able to handle situations like this:
`watson report --from 2014-01-01 --day -y`
And there are quite implicit rules come to a game: user can only guess how the above command resolves options.

So, just because of mutual exclusivity of these options: click doesn't support this feature out of the box, and I decide to choose more explicit solution by using custom class `MutuallyExclusiveOption `.